### PR TITLE
sidecarset support k8s 1.28 sidecarContainers

### PR DESCRIPTION
**Pull request by zmberg**
### Ⅰ. Describe what this PR does
Currently only support inject sidecarContainers.

### CODE REVIEW

File: pkg/control/sidecarcontrol/hash.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ func SidecarSetHashWithoutImage(sidecarSet *appsv1alpha1.SidecarSet) (string, er
 	for i := range ss.Spec.Containers {
 		ss.Spec.Containers[i].Image = ""
 	}
+	for i := range ss.Spec.InitContainers {
+		ss.Spec.InitContainers[i].Image = ""
```
Comment (furykerry) [inline pkg/control/sidecarcontrol/hash.go:47]: plz add ut for init containers in SidecarSetHashWithoutImage

File: pkg/webhook/pod/mutating/sidecarset.go [hunk 2]
```diff
@@ -393,13 +395,22 @@ func buildSidecars(isUpdated bool, pod *corev1.Pod, oldPod *corev1.Pod, matchedS
...
+				} else {
+					sidecarInitContainers = append(sidecarInitContainers, initContainer)
+				}
 			}
 			//process imagePullSecrets
 			sidecarSecrets = append(sidecarSecrets, sidecarSet.Spec.ImagePullSecrets...)
 		}
 
-		sidecarList := sets.NewString()
 		//process containers
```
Comment (furykerry) [inline pkg/webhook/pod/mutating/sidecarset.go:414]: maybe an TODO, implement PodInjectPolicy for initContainers, currently initcontainers only use AfterAppContainer

**Review by furykerry — COMMENTED** [review]
/lgtm

**Comment by zmberg** [comment]
https://github.com/openkruise/kruise/issues/1540

**Pull request merged**